### PR TITLE
Pull s3 config out of model

### PR DESCRIPTION
**Pull request by ersatzryan**
Thoughts? Could use some tests around the `Uploader::Config`, but maybe not

### CODE REVIEW

File: lib/uploader/config.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+module Uploader
+  module Config
```
Comment (laserlemon) [inline lib/uploader/config.rb]: Why two modules? Could this just be `Uploader.options` or `UploaderConfig.options`?

Comment (laserlemon) [inline lib/uploader/config.rb]: Or a `HasPaperclip` module that defines `paperclip_options`:

``` ruby
class Upload < ActiveRecord::Base
  include HasPaperclip

  has_attached_file :file, paperclip_options
end
```

**Comment by laserlemon** [comment]
Another thing to consider is that there will be other "features" that are activated/deactivated based on the presence/absence of `ENV` configuration. Is there a better way to encapsulate those feature switches?

File: lib/uploader/config.rb (hunk 1 above)
Comment (ersatzryan) [inline lib/uploader/config.rb]: cause I was tired and I just wanted to get my thoughts down, cause you said you weren't really following what I was talking about

**Comment by ersatzryan** [comment]
had some time this morning to actually move this around. @laserlemon :thought_balloon: :question: :+1: :arrow_forward: :eggplant:

File: app/models/concerns/has_paperclip.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+module HasPaperclip
+    def paperclip_options
```
Comment (emilford) [inline app/models/concerns/has_paperclip.rb]: 4 spaces.

**Comment by laserlemon** [comment]
:+1: :eggplant: :pray:

**Comment by laserlemon** [comment]
Wait.

File: app/models/concerns/has_paperclip.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    using_s3? ? s3_options : filesystem_options
+  end
+
+  def using_s3?
+    ENV.values_at("S3_BUCKET", "S3_ACCESS_KEY_ID", "S3_SECRET_ACCESS_KEY").all?(&:present?)
+  end
+
+  def s3_options
+    {
+      path: "/uploads/:fingerprint.:extension",
```
Comment (laserlemon) [inline app/models/concerns/has_paperclip.rb:12]: This…

File: app/models/concerns/has_paperclip.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        secret_access_key: ENV["S3_SECRET_ACCESS_KEY"]
+      },
+      s3_protocol: "https",
+      storage: :s3,
+      url: ":s3_domain_url"
+    }
+  end
+
+  def filesystem_options
+    { url: "/uploads/:fingerprint.:extension" }
```
Comment (laserlemon) [inline app/models/concerns/has_paperclip.rb:25]: and this… are model-specific. We may only use this module once but we can at least _feign_ reusability.

Comment (laserlemon) [inline app/models/concerns/has_paperclip.rb:25]: Actually, if we _were_ to reuse these options, the "uploads" naming would still probably be fitting (model naming aside) and the only concern would be MD5 collision (which I'm not concerned about).

**Comment by laserlemon** [comment]
Proceed.

**Pull request merged**